src/docfx/docset/Document.cs

```diff
@@ -154,22 +159,9 @@ internal static ContentType GetContentType(string path, string docsetPath)
             }
         }
 
-        internal static string GetOutputPath(string path, ContentType contentType)
-        {
-            switch (contentType)
-            {
-                case ContentType.Markdown:
-                case ContentType.SchemaDocument:
-                case ContentType.TableOfContents:
-                    return Path.ChangeExtension(path, ".json");
-                default:
-                    return path;
-            }
-        }
-
-        internal static string GetSiteUrl(string path, ContentType contentType)
+        internal static string GetSiteUrl(string path, ContentType contentType, Config config)
         {
-            path = '/' + path;
+            path = '/' + ApplyRoutes(path, config.Routes);
```

Comment:
>path = '/' + ApplyRoutes(path, config.Routes); [](start = 12, length = 46)

there would be a problem when calculating the relative site url in Resolve

Reply:
we'll handle it in url resolve.